From 0536741549a157627d4306f6e19f1cc5cb52cef8 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Sat, 27 Feb 2021 15:03:33 +0000
Subject: [PATCH] CMakeLists.txt: Respect user's libdir

Without this patch, on systems where /usr/lib64/ is used
(e.g. Gentoo), libraries are installed
to /usr/lib/ despite the standard CMake arguments
being passed.

It's not exactly clear what *the* canonical way
is to take libdir parameters in CMake, but
GNUInstallDirs mostly handles it for the *NIX case.

To be minimally invasive here, I've only changed
the *NIX path and aligned as much of the logic as possible.

Signed-off-by: Sam James <sam@gentoo.org>
---
 CMakeLists.txt | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 56b1c49..c4ce5a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -229,7 +229,13 @@ if (OIS_BUILD_DEMOS)
 
 endif()
 
-set(LIB_INSTALL_DIR "lib")
+if(UNIX)
+	include(GNUInstallDirs)
+	set(LIB_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR})
+else()
+	set(LIB_INSTALL_DIR "lib")
+endif()
+
 set(OIS_LIB_DIRECTORY ${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR})
 
 file(GLOB PUBLIC_HEADERS "${CMAKE_CURRENT_SOURCE_DIR}/includes/*.h")
@@ -240,8 +246,8 @@ install(FILES ${PUBLIC_HEADERS}
 
 install(TARGETS OIS
   RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+  ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
   FRAMEWORK DESTINATION Library/Frameworks
 )
 
-- 
2.26.2

